File: tests/conformance/fixtures/go/18-edge-provides-empty.go
package main

import sykli "github.com/yairfalse/sykli/sdk/go"

func main() {
	p := sykli.New()

	// provides with empty string value should omit value field
	p.Task("build").Run("make build").Provides("artifact", "")

	// provides with no value should also omit value field
	p.Task("migrate").Run("dbmate up").Provides("db-ready")

	// provides with actual value should include it
	p.Task("package").Run("docker build").Provides("image", "myapp:latest").After("build")

	p.Emit()
}

File: tests/conformance/fixtures/elixir/18-edge-provides-empty.exs
use Sykli

pipeline do
  task "build" do
    run "make build"
    provides "artifact"
  end

  task "migrate" do
    run "dbmate up"
    provides "db-ready"
  end

  task "package" do
    run "docker build"
    after_ ["build"]
    provides "image", "myapp:latest"
  end
end
|> Sykli.Emitter.validate!()
|> Sykli.Emitter.to_json()
|> IO.puts()
